access_keys_test.go

```diff
@@ -1,77 +1,62 @@
+// Package walletclient here we are testing walletclient public methods
 package walletclient
 
 import (
 	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
-	"github.com/bitcoin-sv/spv-wallet/models"
-	"github.com/stretchr/testify/assert"
-
 	"github.com/bitcoin-sv/spv-wallet-go-client/fixtures"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/stretchr/testify/require"
 )
 
 // TestAccessKeys will test the AccessKey methods
 func TestAccessKeys(t *testing.T) {
-	transportHandler := testTransportHandler{
-		Type:      fixtures.RequestType,
-		Path:      "/access-key",
-		Result:    fixtures.MarshallForTestHandler(fixtures.AccessKey),
-		ClientURL: fixtures.ServerURL,
-		Client:    WithHTTPClient,
-	}
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/v1/access-key":
+			if r.Method == http.MethodGet {
+				json.NewEncoder(w).Encode(fixtures.AccessKey)
+			} else if r.Method == http.MethodPost {
+				json.NewEncoder(w).Encode(fixtures.AccessKey)
+			} else if r.Method == http.MethodDelete {
+				json.NewEncoder(w).Encode(fixtures.AccessKey)
+			}
```

Comment:
In these three `if` blocks there is the same code - consider simplifying this.

Reply:
well it is quite simplified as it unit test ... but right ..